### PR TITLE
bind-mount support

### PR DESCRIPTION
**Pull request by david-piiano**
And also added temporary directory to host these bind mounts

### CODE REVIEW

File: cli/flows-cli.sh [hunk 1]
```diff
@@ -226,6 +227,34 @@ create_gradle_volume() {
   fi
 }
 
+create_m2_bind_mount() {
+  M2_BM_FOLDER=${FLOWS_TEMP_FOLDER}/m2
+  if [ -d ${M2_BM_FOLDER} ] ; then
+    echo "[ ] Reusing bind mount to ${M2_BM_FOLDER}. (to remove: rm -r ${M2_BM_FOLDER})"
```
Comment (nerya10) [inline cli/flows-cli.sh:233]: what it mean to reuse a bind mount ? Isnt it just a mapping between the container and directy on the host? does it saved after the container is exit ?

File: cli/README.md [hunk 2]
```diff
@@ -65,7 +65,8 @@ This script will run the Piiano Flows container and analyze the code you provide
...
 2. `FLOWS_SCAN_NAME`(defaults to the local directory name being scanned) - only relevant when `PIIANO_CS_VIEWER_MODE` is `online`. Sets the scan name for the online viewer.
 3. `FLOWS_PORT`(3000 by default) - The flows viewer will run on this port unless it is already taken. In that case the script will automatically find the next available port.
-4. `FLOWS_USE_VOLUMES` (`true` by default) - normally, the script will create volumes for Maven and Gradle if they don't already exist and map them to the scanner engine (see more configuration in the next section). When set to `false`, the script will not attempt to create or map any volumes.
+4. `FLOWS_MOUNT_TYPE` (`volume` by default) - normally, the script will create docker volumes for Maven and Gradle if they don't already exist and map them to the scanner engine (see more configuration in the next section). Alternatively you can use bind mounts instead of docker volumes by setting the variable to `bind-mount`. When set to `none`, the script will not attempt to create or map any volumes.
+5. `FLOWS_TEMP_FOLDER` (`/tmp` by default) -
```
Comment (nerya10) [inline cli/README.md:69]: use with bind mount

File: cli/flows-cli.sh [hunk 3]
```diff
@@ -226,6 +227,42 @@ create_gradle_volume() {
   fi
 }
 
+create_m2_bind_mount() {
+  M2_BM_FOLDER=${FLOWS_TEMP_FOLDER}/m2
```
Comment (nerya10) [inline cli/flows-cli.sh:231]: ${FLOWS_TEMP_FOLDER}/.m2 or ${FLOWS_TEMP_FOLDER}/m2 ?

File: cli/flows-cli.sh [hunk 4]
```diff
@@ -226,6 +227,42 @@ create_gradle_volume() {
...
+    cp -R ${PIIANO_CS_M2_FOLDER} ${M2_BM_FOLDER}
+  fi
+  
+  # Count directories using find command and count lines
+  num_directories=$(find "${M2_BM_FOLDER}" -mindepth 1 -type d | wc -l)
+  echo "Number of directories in ${M2_BM_FOLDER}: $num_directories"
+}
+
+create_gradle_bind_mount() {
+  GRADLE_BM_FOLDER=${FLOWS_TEMP_FOLDER}/gradle
```
Comment (nerya10) [inline cli/flows-cli.sh:249]: .gradle ?

File: cli/flows-cli.sh (hunk 3 above)
Comment (david-piiano) [reply cli/flows-cli.sh:231]: I don't see value making it `.m2`, this is internal so `m2` is fine

File: cli/flows-cli.sh (hunk 4 above)
Comment (david-piiano) [reply cli/flows-cli.sh:249]: gradle without dot is fine here, this is internal

**Review by nerya10 — APPROVED** [review]

**Pull request merged**